closed/OpenJ9.gmk

```diff
@@ -454,14 +454,47 @@ ifneq (,$(OPENJ9_DEVELOPER_DIR))
 endif
 
 ifeq (true,$(OPENJ9_ENABLE_CMAKE))
+CMAKE_ARGS :=
+
+# We grab the C/C++ compilers detected by autoconf or provided by user, not
+# the CC/CXX variables defined by the makefiles, which potentitally include
+# the ccache command which will throw off cmake
+ifneq (,$(OPENJ9_CC))
+  CMAKE_ARGS += "-DCMAKE_C_COMPILER=$(OPENJ9_CC)"
+else
+  CMAKE_ARGS += "-DCMAKE_C_COMPILER=$(ac_cv_prog_CC)"
+endif
+
+ifneq (,$(OPENJ9_CXX))
+  CMAKE_ARGS += "-DCMAKE_CXX_COMPILER=$(OPENJ9_CXX)"
+else
+  CMAKE_ARGS += "-DCMAKE_CXX_COMPILER=$(ac_cv_prog_CXX)"
+endif
+
+ifneq (,$(CCACHE))
+  # Open jdk makefiles add a  bunch of environemnt variables to the ccache command
+  # cmake will not parse this properly, so we wrap the whole thing in the env command
+  # We also need to add semicolons between arguments or else cmake will treat the whole
+  # thing as one long command name
+
+  # Note: we remove the CCACHE_COMPRESS option that openjdk adds, because it significantly
+  # slows down the build (to the point of erasing any gains from using ccache)
+  CCACHE_NOCOMPRESS := $(filter-out CCACHE_COMPRESS=1,$(CCACHE))
+  ESCAPED_CCACHE :=env$(shell printf ";%s" $(CCACHE_NOCOMPRESS))
```

Comment:
Why do we need the shell to do the equivalent of `env;$(CCACHE_NOCOMPRESS)`?

Reply:
printf actually repeats the format specifier as long as there are remaining arguments. So if `CCACHE_NOCOMPRESS= foo bar baz` we get `ESCAPED_CCACHE=env;foo;bar;baz`. We need to do this to put it in a list format for cmake, otherwise it will treat it as one long command name with embedded spaces.

Reply:
Then I think you want:
```
  ESCAPED_CCACHE := env$(subst $(SPACE),,$($(addprefix ;,$(CCACHE_NOCOMPRESS)))
```